modules/apps/batch-planner/batch-planner-rest-test/src/testIntegration/java/com/liferay/batch/planner/rest/resource/v1_0/test/FieldResourceTest.java

```diff
@@ -6,14 +6,267 @@
+import com.liferay.object.field.util.ObjectFieldUtil;
+import com.liferay.object.model.ObjectDefinition;
+import com.liferay.object.model.ObjectRelationship;
+import com.liferay.object.rest.test.util.ObjectRelationshipTestUtil;
+import com.liferay.object.service.ObjectFieldLocalServiceUtil;
+import com.liferay.object.test.util.ObjectDefinitionTestUtil;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.util.ListUtil;
+import com.liferay.portal.kernel.util.StringUtil;
 
+import java.util.Collections;
+import java.util.Objects;
+
+import org.junit.Assert;
 import org.junit.Ignore;
+import org.junit.Test;
 import org.junit.runner.RunWith;
 
 /**
  * @author Matija Petanjek
  */
-@Ignore
 @RunWith(Arquillian.class)
 public class FieldResourceTest extends BaseFieldResourceTestCase {
+
+	@Test
+	public void testGetFieldsWithManyToManyRelationship() throws Exception {
+		String fieldName = "x" + RandomTestUtil.randomString();
+
+		ObjectDefinition objectDefinition1 =
+			ObjectDefinitionTestUtil.publishObjectDefinition(
+				ObjectDefinitionTestUtil.getRandomName(),
+				Collections.singletonList(
+					ObjectFieldUtil.createObjectField(
+						ObjectFieldConstants.BUSINESS_TYPE_TEXT,
+						ObjectFieldConstants.DB_TYPE_STRING, true, true, null,
+						RandomTestUtil.randomString(), fieldName, false)),
+				ObjectDefinitionConstants.SCOPE_COMPANY);
+
+		ObjectDefinition objectDefinition2 =
+			ObjectDefinitionTestUtil.publishObjectDefinition(
+				ObjectDefinitionTestUtil.getRandomName(),
+				Collections.singletonList(
+					ObjectFieldUtil.createObjectField(
+						ObjectFieldConstants.BUSINESS_TYPE_TEXT,
+						ObjectFieldConstants.DB_TYPE_STRING, true, true, null,
+						RandomTestUtil.randomString(),
+						"x" + RandomTestUtil.randomString(), false)),
+				ObjectDefinitionConstants.SCOPE_COMPANY);
+
+		ObjectRelationship objectRelationship =
+			ObjectRelationshipTestUtil.addObjectRelationship(
+				objectDefinition1, objectDefinition2,
+				TestPropsValues.getUserId(),
+				ObjectRelationshipConstants.TYPE_MANY_TO_MANY);
+
+		Page<Field> page = fieldResource.getPlanInternalClassNameKeyFieldsPage(
+			_getObjectDefinitionInternalClassName(objectDefinition1.getName()),
+			null);
+
+		assertEqualsIgnoringOrder(
+			ListUtil.fromArray(
+				_toField(null, "externalReferenceCode", false, "string", null),
+				_toField(null, "keywords", false, "array", "CSV"),
+				_toField(null, "permissions", false, "array", null),
+				_toField(null, "taxonomyCategoryIds", false, "array", "CSV"),
+				_toField(null, fieldName, false, "string", null),
+				_toField(
+					null, objectRelationship.getName(), false, "array", null)),
+			ListUtil.fromCollection(page.getItems()));
+	}
+
+	@Test
+	public void testGetFieldsWithMultipleOneToManyRelationship()
+		throws Exception {
+
+		ObjectDefinition objectDefinition1 =
```

Comment:
Better to test from the objectDefinition1 point of view. I think it simplifies the test and make it more understanding